xml/tuning_cgroups.xml

```diff
@@ -286,246 +286,225 @@ TasksMax=16284
+<para>
+You can apply the values to (long running) services permanently.
+</para>
 <screen>
-&prompt.reader1;sync; echo 3 > /proc/sys/vm/drop_caches
-&prompt.reader1;echo $$; dd if=file1 of=/dev/null bs=4k
-5633
-[...]
-&prompt.reader2;echo $$; dd if=file2 of=/dev/null bs=4k count=131072
-5703
-[...]
+&prompt.sudo;<command>systemctl set-property fast.service IOWeight=400</command>
+&prompt.sudo;<command>systemctl set-property slow.service IOWeight=50</command>
+&prompt.sudo;<command>systemctl set-property throttled.service IOReadBandwidthMax="/dev/sda 1M"</command>
 </screen>
-   <para>
-    Try doubling the weight again for reader2, and testing the new setting:
-   </para>
+
+<para>
+Alternatively, you can apply I/O control to individually run commands, for instance:
+</para>
 <screen>
-&prompt.reader2;echo 400 > reader1/blkio.bfq.weight
-&prompt.reader2;cat reader2/blkio.bfq.weight
-400
-&prompt.reader1;sync; echo 3 > /proc/sys/vm/drop_caches
-&prompt.reader1;echo $$; dd if=file1.txt of=/dev/null bs=4k
-[...]
-&prompt.reader2;echo $$; dd if=file2.txt of=/dev/null bs=4k count=131072
-[...]
+&prompt.sudo;<command>systemd-run --scope -p IOWeight=400 prioritized_command</command>
+&prompt.sudo;<command>systemd-run --scope -p IOWeight=50  low_prioritized_command</command>
+&prompt.sudo;<command>systemd-run --scope -p IOReadBandwidthMax="/dev/sda 1M" dd if=/dev/sda of=/dev/null bs=1M count=10</command>
 </screen>
-  </sect2>
- </sect1>
+</sect2>
+
+<sect2 xml:id="sec-cgroups-control-behavior">
+<title>I/O control behavior and setting expectations</title>
+<para>
+ The following list items describe I/O control behavior, and what you
+ should expect under various conditions.
+</para>
+<itemizedlist>
+<listitem>
+<para>
+I/O control works best for direct I/O operations (bypassing page cache),
+the situations where the actual IO is decoupled from the caller (typically
+writeback via page cache) may manifest variously – delayed I/O control or
+even no observed I/O control (consider little bursts or competing
+workloads that happen to never "meet" submitting I/O at the same time (and
+hence saturating the bandwidth). For these reasons, the resulting ratio of
+I/O throughputs does not strictly follow the ratio of configured weights.
+</para>
+</listitem>
+<listitem>
+<para>
+systemd performs scaling of configured weights (to adjust for narrower BFQ
+weight range), hence the resulting throughput ratios also differ.
+</para>
+</listitem>
+<listitem>
+<para>
+The writeback activity depends on the amount of dirty pages, besides the
+global sysctl knobs (<filename>vm.dirty_background_ratio</filename> and
+<filename>vm.dirty_ratio</filename>) memory limits of individual cgroups
+come into play when the dirty limits are distributed among cgroups and
+this in turn may affect I/O intensity of affected cgroups.
+</para>
+</listitem>
+<listitem>
+<para>
+As explained in the introduction, inodes are the unit of writeback charging,
```

Comment:
The introduction lost the inodes explanation, so this is a broken "reference". I think "As explained in the introduction" can be dropped here too.